### PR TITLE
expression: implement vectorized evaluation for builtinFromUnixTime2ArgSig

### PR DESCRIPTION
**Pull request by ekalinin**
PCP #12106

### What problem does this PR solve? 

This PR implements vectorized `builtinFromUnixTime2ArgSig`

### What is changed and how it works?

```sh
➜ make vectorized-bench VB_FILE=String VB_FUNC=builtinFromUnixTime2ArgSig
cd ./expression && \
	go test -v -benchmem \
		-bench=BenchmarkVectorizedBuiltinStringFunc \
		-run=BenchmarkVectorizedBuiltinStringFunc \
		-args "builtinFromUnixTime2ArgSig"
[2019/11/09 22:45:29.543 +03:00] [WARN] [builtin_string.go:268] ["unexpected `Flen` value(-1) in CONCAT's args"] ["arg's index"=0]
[2019/11/09 22:45:29.543 +03:00] [WARN] [builtin_string.go:268] ["unexpected `Flen` value(-1) in CONCAT's args"] ["arg's index"=1]
[2019/11/09 22:45:29.543 +03:00] [WARN] [builtin_string.go:268] ["unexpected `Flen` value(-1) in CONCAT's args"] ["arg's index"=2]
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinStringFunc/builtinFromUnixTime2ArgSig-VecBuiltinFunc-12         	    3189	   1356461 ns/op	   85537 B/op	    1941 allocs/op
BenchmarkVectorizedBuiltinStringFunc/builtinFromUnixTime2ArgSig-NonVecBuiltinFunc-12      	     766	   1406440 ns/op	   85536 B/op	    1941 allocs/op
PASS
ok  	github.com/pingcap/tidb/expression	5.655s
```

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: expression/builtin_time_vec.go [hunk 1]
```diff
@@ -128,11 +128,56 @@ func (b *builtinDateSig) vectorized() bool {
...
+	defer b.bufAllocator.put(buf2)
+	if err = b.args[1].VecEvalString(b.ctx, input, buf2); err != nil {
+		return err
+	}
+
+	result.ReserveString(n)
+	ds := buf1.Decimals()
+	fsp := int8(b.tp.Decimal)
+	for i := 0; i < n; i++ {
+		if buf1.IsNull(i) {
```
Comment (b41sh) [inline expression/builtin_time_vec.go]: can we use `if buf1.IsNull(i) || buf2.IsNull(i) {` here?

File: expression/builtin_string_vec_test.go [hunk 2]
```diff
@@ -225,6 +225,9 @@ var vecBuiltinStringCases = map[string][]vecExprBenchCase{
 	ast.Elt: {
 		{retEvalType: types.ETString, childrenTypes: []types.EvalType{types.ETInt, types.ETString, types.ETString, types.ETString}, geners: []dataGenerator{&rangeInt64Gener{-1, 5}}},
 	},
+	ast.FromUnixTime: {
+		{retEvalType: types.ETString, childrenTypes: []types.EvalType{types.ETDecimal, types.ETString}},
```
Comment (SunRunAway) [inline expression/builtin_string_vec_test.go]: Should we use a specific generator for the second argument?

Comment (SunRunAway) [inline expression/builtin_string_vec_test.go]: Should we use a specific generator for the second argument?

File: expression/builtin_time_vec.go (hunk 1 above)
Comment (ekalinin) [reply expression/builtin_time_vec.go]: Sure. Fixed.

File: expression/builtin_string_vec_test.go (hunk 2 above)
Comment (ekalinin) [reply expression/builtin_string_vec_test.go]: Fixed

**Comment by b41sh** [comment]
LGTM

**Pull request merged**